code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,388 @@
+	if(unlink)
+		cell = null
+	else
+		cell = new(src)
+
+/obj/item/melee/secsword/emp_act(severity)
+	cell.use(round(cell.charge / severity))
+	update_icon()
+
+/obj/item/melee/secsword/get_cell()
+	return cell
+
+/obj/item/melee/secsword/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/stock_parts/cell))
+		return
+	var/obj/item/stock_parts/cell/C = I
+	if(cell)
+		to_chat(user, "<span class='warning'>[src] already has a cell!</span>")
+		return
+	if(C.maxcharge < stam_hitcost)
+		to_chat(user, "<span class='warning'>[src] requires a higher capacity cell!</span>")
+		return
+	if(!user.unEquip(I))
+		return
+	I.forceMove(src)
+	cell = I
+	to_chat(user, "<span class='notice'>You install [I] into [src].</span>")
+	update_icon()
+
+/obj/item/melee/secsword/screwdriver_act(mob/living/user, obj/item/I)
+	if(!cell)
+		to_chat(user, "<span class='warning'>There's no cell installed!</span>")
+		return
+	if(!I.use_tool(src, user, volume = I.tool_volume))
+		return
+
+	user.put_in_hands(cell)
+	to_chat(user, "<span class='notice'>You remove [cell] from [src].</span>")
+	cell.update_icon()
+	cell = null
+	update_icon()
+
+/obj/item/melee/secsword/attack_self(mob/user)
+	if(!cell)
+		to_chat(user, "<span class='warning'>[src] does not have a power source!</span>")
+		return
+
+	if(state == SECSWORD_OFF)
+		if(cell.charge < stam_hitcost)
+			state = SECSWORD_OFF
+			to_chat(user, "<span class='notice'>[src] does not have enough charge to stun!</span>")
+
+		else
+			state = SECSWORD_STUN
+			to_chat(user, "<span class='warning'>[src]'s edge is now set to stun.</span>")
+	else if(state == SECSWORD_STUN)
+		if(cell.charge < burn_hitcost)
+			state = SECSWORD_OFF
+			to_chat(user, "<span class='notice'>[src] does not have enough charge to burn!</span>")
+		else
+			state = SECSWORD_BURN
+			to_chat(user, "<span class='warning'>[src]'s edge is now set to burn.</span>")
+	else
+		state = SECSWORD_OFF
+		to_chat(user, "<span class='notice'>[src]'s edge is now turned off.</span>")
+	update_icon()
+	add_fingerprint(user)
+
+/obj/item/melee/secsword/attack(mob/M, mob/living/user)
+	if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+		if(state == SECSWORD_STUN && sword_stun(user, user, skip_cooldown = TRUE))
+			user.visible_message("<span class='danger'>[user] accidentally hits [user.p_themselves()] with [src]!</span>",
+							"<span class='userdanger'>You accidentally hit yourself with [src]!</span>")
+		return
+	if(user.mind?.martial_art?.no_baton && user.mind?.martial_art?.can_use(user)) // Just like the baton, no sword + judo.
+		to_chat(user, "<span class='warning'>The sword feels off-balance in your hand due to your specific martial training!</span>")
+		return
+
+	if(state == SECSWORD_OFF) // Off or no battery
+		return ..()
+	else if(state == SECSWORD_STUN) // Stamina
+		if(issilicon(M)) // Can't stun borgs and AIs
+			return ..()
+		if(!isliving(M))
+			return ..()
+		var/mob/living/L = M
+		sword_stun(L, user)
```

Comment:
```suggestion
		sword_stun(M, user)
```
no point in typecasting if you're just going to immediately pass it as an argument